File: frontend-admin/apps/web-antd/src/api/system/storage-bucket.ts
import { requestClient } from '#/api/request';

export namespace StorageBucketApi {
  export interface StorageBucket {
    id: number;
    name: string;
    driver: 'local' | 'minio' | 'oss' | 'cos';
    endpoint: string;
    bucket: string;
    accessKey: string;
    secretKey: string;
    region: string;
    useSsl: boolean;
    cdnDomain: string;
    pathPrefix: string;
    purpose: string;
    isDefault: boolean;
    status: number;
    description: string;
    createdAt: string;
    updatedAt: string;
  }

  export interface CreateStorageBucket {
    name: string;
    driver: string;
    endpoint?: string;
    bucket?: string;
    accessKey?: string;
    secretKey?: string;
    region?: string;
    useSsl?: boolean;
    cdnDomain?: string;
    pathPrefix?: string;
    purpose?: string;
    isDefault?: boolean;
    status?: number;
    description?: string;
  }
}

/**
 * 获取所有存储桶
 */
export function getAllStorageBucketsApi() {
  return requestClient.get<StorageBucketApi.StorageBucket[]>(
    '/system/storage-buckets',
  );
}

/**
 * 根据ID获取存储桶
 */
export function getStorageBucketByIdApi(id: number) {
  return requestClient.get<StorageBucketApi.StorageBucket>(
    `/system/storage-buckets/${id}`,
  );
}

/**
 * 创建存储桶
 */
export function createStorageBucketApi(
  data: StorageBucketApi.CreateStorageBucket,
) {
  return requestClient.post('/system/storage-buckets', data);
}

/**
 * 更新存储桶
 */
export function updateStorageBucketApi(
  id: number,
  data: StorageBucketApi.CreateStorageBucket,
) {
  return requestClient.put(`/system/storage-buckets/${id}`, data);
}

/**
 * 删除存储桶
 */
export function deleteStorageBucketApi(id: number) {
  return requestClient.delete(`/system/storage-buckets/${id}`);
}

/**
 * 设置默认存储桶
 */
export function setDefaultStorageBucketApi(id: number) {
  return requestClient.put(`/system/storage-buckets/${id}/default`);
}

/**
 * 测试存储桶连接
 */
export function testStorageBucketConnectionApi(id: number) {
  return requestClient.post<{ message: string }>(
    `/system/storage-buckets/${id}/test`,
  );
}

/**
 * 获取已启用的存储驱动列表
 */
export function getEnabledDriversApi() {
  return requestClient.get<
    Array<{ value: string; label: string; icon: string; enabled: boolean }>
  >('/system/storage-buckets/enabled-drivers');
}

/**
 * 按驱动和桶名测试连接（无需先保存）
 */
export function testStorageBucketByDriverApi(data: {
  driver: string;
  bucketName: string;
  region?: string;
}) {
  return requestClient.post<{ message: string }>(
    '/system/storage-buckets/test-by-driver',
    data,
  );
}

File: backend-server/internal/model/storage_bucket.go
package model

import "time"

// StorageBucket 存储桶配置模型
type StorageBucket struct {
	ID         int64     `json:"id" gorm:"primaryKey;autoIncrement"`
	Name       string    `json:"name" gorm:"size:100;not null;uniqueIndex:uk_name"`
	Driver     string    `json:"driver" gorm:"size:20;not null;index:idx_driver"`
	Endpoint   string    `json:"endpoint" gorm:"size:500"`
	Bucket     string    `json:"bucket" gorm:"size:200"`
	AccessKey  string    `json:"accessKey" gorm:"size:500"`
	SecretKey  string    `json:"secretKey" gorm:"size:500"`
	Region     string    `json:"region" gorm:"size:100"`
	UseSSL     bool      `json:"useSsl" gorm:"default:false"`
	CDNDomain  string    `json:"cdnDomain" gorm:"size:500"`
	PathPrefix string    `json:"pathPrefix" gorm:"size:500"`
	Purpose    string    `json:"purpose" gorm:"size:100;index:idx_purpose"`
	IsDefault  bool      `json:"isDefault" gorm:"default:false"`
	Status     int8      `json:"status" gorm:"default:1;index:idx_status"`
	Description string   `json:"description" gorm:"size:500"`
	CreatedAt  time.Time `json:"createdAt" gorm:"autoCreateTime"`
	UpdatedAt  time.Time `json:"updatedAt" gorm:"autoUpdateTime"`
}

func (StorageBucket) TableName() string {
	return "sys_storage_bucket"
}
